src/components/navbars/DesktopNav.jsx

```diff
@@ -13,24 +14,33 @@ const DesktopNav = ({
   isJoin,
 }) => {
   const { isLoggedIn } = useAuth();
-  // const {
-  //   classes,
-  //   currentPage,
-  //   totalPages,
-  //   handleSearch,
-  //   handlePageChange,
-  //   setCategory,
-  // } = useSearch();
+  const [searchTerm, setSearchTerm] = useState('');
+  const navigate = useNavigate();
+
+  const handleSearch = (event) => {
+    event.preventDefault();
+    if (searchTerm.trim()) {
+      navigate(`/search?page=1&query=${searchTerm}`);
```

Comment:
BUG: when you try to go back a page after being redirected to the search page you cannot easily return to the previous page by hitting "go back" in the browser. There are a few changes in the URL before it finally redirects to the homepage.

Reply:
I tried to fix but not sure if it works like it should be